pkg/provisioner/filestorage/filestorage.go

```diff
@@ -0,0 +1,218 @@
+	"k8s.io/client-go/pkg/api/v1"
+
+	"github.com/oracle/oci-volume-provisioner/pkg/oci/client"
+	"github.com/oracle/oci-volume-provisioner/pkg/provisioner/plugin"
+)
+
+const (
+	ociVolumeID            = "ociVolumeID"
+	ociExportID            = "ociExportID"
+	volumePrefixEnvVarName = "OCI_VOLUME_NAME_PREFIX"
+	fsType                 = "fsType"
+	subnetID               = "subnetId"
+	mntTargetID            = "mntTargetId"
+)
+
+// filesystemProvisioner is the internal provisioner for OCI filesystem volumes
+type filesystemProvisioner struct {
+	client client.ProvisionerClient
+}
+
+var _ plugin.ProvisionerPlugin = &filesystemProvisioner{}
+
+// NewFilesystemProvisioner creates a new file system provisioner that creates
+// filsystems using OCI file system service.
+func NewFilesystemProvisioner(client client.ProvisionerClient) plugin.ProvisionerPlugin {
+	return &filesystemProvisioner{
+		client: client,
+	}
+}
+
+// getMountTargetFromID retrieves mountTarget from given mountTargetID
+func getMountTargetFromID(ctx context.Context, mountTargetID string, fileStorageClient client.FileStorage) *filestorage.MountTarget {
+	responseMnt, err := fileStorageClient.GetMountTarget(ctx, filestorage.GetMountTargetRequest{
+		MountTargetId: common.String(mountTargetID),
+	})
+	if err != nil {
+		glog.Errorf("Failed to retrieve mount point: %s", err)
+		return nil
+	}
+	return &responseMnt.MountTarget
+}
+
+func (filesystem *filesystemProvisioner) Provision(
+	options controller.VolumeOptions,
+	availabilityDomain *identity.AvailabilityDomain) (*v1.PersistentVolume, error) {
+
+	ctx, cancel := context.WithTimeout(filesystem.client.Context(), filesystem.client.Timeout())
+	defer cancel()
+
+	fileStorageClient := filesystem.client.FileStorage()
+	response, err := fileStorageClient.CreateFileSystem(ctx, filestorage.CreateFileSystemRequest{
+		CreateFileSystemDetails: filestorage.CreateFileSystemDetails{
+			AvailabilityDomain: availabilityDomain.Name,
+			CompartmentId:      common.String(filesystem.client.CompartmentOCID()),
+			DisplayName:        common.String(fmt.Sprintf("%s%s", os.Getenv(volumePrefixEnvVarName), options.PVC.Name)),
+		},
+	})
+	if err != nil {
+		glog.Errorf("Failed to create a file system storage:%#v, %s", options, err)
+		return nil, err
+	}
+
+	mntTargetResp := filestorage.MountTarget{}
+	if options.Parameters[mntTargetID] == "" {
+		// Check if there there already is a mount target in the existing compartment
+		glog.Infof("Looking up existing mount targets")
+		responseListMnt, err := fileStorageClient.ListMountTargets(ctx, filestorage.ListMountTargetsRequest{
+			AvailabilityDomain: availabilityDomain.Name,
+			CompartmentId:      common.String(filesystem.client.CompartmentOCID()),
+		})
+		if err != nil {
+			glog.Errorf("Failed to list mount targets:%#v, %s", options, err)
+			return nil, err
+		}
+		if len(responseListMnt.Items) != 0 {
+			glog.Infof("Found mount targets to use")
+			rand.Seed(time.Now().Unix())
```

Comment:
Should seed rand once in `main()` / `init()` rather than on each call.

Reply:
done